components/vocabularies/api/src/main/java/org/phenotips/vocabulary/internal/solr/SolrVocabularyTerm.java

```diff
@@ -130,6 +130,17 @@ public String getId()
     }
 
     @Override
+    public String getUnderlyingId()
+    {
+        if ("hgnc".equals(this.ontology.getIdentifier())) {
+            @SuppressWarnings("unchecked")
+            List<String> ensemblList = (List<String>) this.get("ensembl_gene_id");
+            return ensemblList != null && !ensemblList.isEmpty() ? ensemblList.get(0) : (String) this.get("symbol");
```

Comment:
I believe the SolrVocubularyTerm should be abstracted from the specific vocabularies as much as possible.